tests/components/vesync/test_select.py

```diff
@@ -0,0 +1,73 @@
+"""Tests for the select platform."""
+
+from unittest.mock import patch
+
+from homeassistant.components.select import (
+    ATTR_OPTION,
+    DOMAIN as SELECT_DOMAIN,
+    SERVICE_SELECT_OPTION,
+)
+from homeassistant.components.vesync.const import NIGHT_LIGHT_LEVEL_DIM
+from homeassistant.components.vesync.select import HA_TO_VS_NIGHT_LIGHT_LEVEL_MAP
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+
+from .common import ENTITY_HUMIDIFIER_300S_NIGHT_LIGHT_SELECT
+
+
+async def test_set_nightlight_level(
+    hass: HomeAssistant, config_entry: ConfigEntry, humidifier_300s, manager
+) -> None:
+    """Test set of night light level."""
+
+    with (
+        patch(
+            "homeassistant.components.vesync.async_generate_device_list",
+            return_value=[humidifier_300s],
+        ),
+        patch(
+            "homeassistant.helpers.update_coordinator.DataUpdateCoordinator.async_request_refresh"
```

Comment:
I think we can avoid patching the internals and patch the vesync api instead here. maybe `VeSync.update_all_devices`?

Reply:
The Vesync manager is already being mocked so the mocked device can simply be stuffed into the device list.

Reply:
Changes not ready yet. I will address it tonight.